components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2478,6 +2486,113 @@ private void validateAndSetTransports(APIProduct apiProduct) throws APIManagemen
+                            validatedPolicies.add(policy);
+                        }
+                    } else {
+                        //TODO: get policy based on the name
+                        OperationPolicyData commonPolicyData =
+                                getCommonOperationPolicyByPolicyId(policyId, tenantDomain, false);
+                        if (commonPolicyData != null) {
+                            // A common policy is found for specified policy. This will be validated according to the provided
+                            // attributes and added to API policy list
+                            if (log.isDebugEnabled()) {
+                                log.debug(
+                                        "A common policy is found for " + policyId + ". Validating the policy");
+                            }
+                            OperationPolicySpecification commonPolicySpec = commonPolicyData.getSpecification();
+                            if (validateAppliedPolicyWithSpecification(commonPolicySpec, policy)) {
+                                validatedPolicies.add(policy);
+                            }
+                        } else {
+                            throw new APIManagementException("Selected policy " + policyId + " is not found.",
+                                    ExceptionCodes.INVALID_OPERATION_POLICY);
+                        }
+                    }
+                }
+            }
+            uriTemplate.setOperationPolicies(validatedPolicies);
+        }
+    }
+
+    private boolean validateAppliedPolicyWithSpecification(OperationPolicySpecification policySpecification,
+                                                           OperationPolicy appliedPolicy)
+            throws APIManagementException {
+
+        //Validate the policy applied direction
+        if (!policySpecification.getApplicableFlows().contains(appliedPolicy.getDirection())) {
+            if (log.isDebugEnabled()) {
+                log.debug("The policy " + policySpecification.getName()
+                        + " is not support in the " + appliedPolicy.getDirection() + " flow. Hence skipped.");
+            }
+            throw new APIManagementException(policySpecification.getName() + " cannot be used in the "
+                    + appliedPolicy.getDirection() + " flow.",
+                    ExceptionCodes.OPERATION_POLICY_NOT_ALLOWED_IN_THE_APPLIED_FLOW);
+        }
+
+        //Validate policy Attributes
+        if (policySpecification.getPolicyAttributes() != null) {
+            for (OperationPolicySpecAttribute attribute : policySpecification.getPolicyAttributes()) {
+                if (attribute.isRequired()) {
+                    Object policyAttribute = appliedPolicy.getParameters().get(attribute.getName());
+                    if (policyAttribute != null) {
+                        //TODO: Attribute Type validation is required
```

Comment:
Did we miss the todo logic here?

Reply:
Since it is a todo, we shall look into this later